pkg/auth/webapp/session_authflow.go

```diff
@@ -105,6 +105,9 @@ func newXStep() string {
 // Some steps in an authflow can have branches.
 // In order to be able to switch between branches, we need to remember the state that has branches.
 type AuthflowScreen struct {
+	// Store FlowType and FinishRedirectURI when the flow is finish
+	FlowType          authflow.FlowType `json:"flow_type,omitempty"`
+	FinishRedirectURI string            `json:"finish_redirect_uri,omitempty"`
```

Comment:
Why we don't simply derive finish_redirect_uri in `Finish`?

Reply:
As `finish_redirect_uri` is derived from `*authflow.FlowResponse` but `*authflow.FlowResponse` can't be stringify because `*authflow.FlowResponse.Action.Data` is a interface instead of struct. So we need to derive and save `finish_redirect_uri` from previous screen.

Reply:
The problem here is:
1. The flow will be deleted on finish, so it is not gettable in the service.
1. If we try to store the data in session, we are unable to recover the data from the serialized json, because we don't know the type of data.